std/special/compiler_rt/comparesf2_test.zig

```diff
@@ -0,0 +1,101 @@
+// Ported from:
+//
+// https://github.com/llvm/llvm-project/commit/d674d96bc56c0f377879d01c9d8dfdaaa7859cdb/compiler-rt/test/builtins/Unit/comparesf2_test.c
+
+const std = @import("std");
+const builtin = @import("builtin");
+const is_test = builtin.is_test;
+
+const comparesf2 = @import("comparesf2.zig");
+
+const TestVector = struct {
+    a: f32,
+    b: f32,
+    eqReference: c_int,
+    geReference: c_int,
+    gtReference: c_int,
+    leReference: c_int,
+    ltReference: c_int,
+    neReference: c_int,
+    unReference: c_int,
+};
+
+fn test__cmpsf2(vector: TestVector) bool {
+    if (comparesf2.__eqsf2(vector.a, vector.b) != vector.eqReference) {
+        return false;
+    }
+    if (comparesf2.__gesf2(vector.a, vector.b) != vector.geReference) {
+        return false;
+    }
+    if (comparesf2.__gtsf2(vector.a, vector.b) != vector.gtReference) {
+        return false;
+    }
+    if (comparesf2.__lesf2(vector.a, vector.b) != vector.leReference) {
+        return false;
+    }
+    if (comparesf2.__ltsf2(vector.a, vector.b) != vector.ltReference) {
+        return false;
+    }
+    if (comparesf2.__nesf2(vector.a, vector.b) != vector.neReference) {
+        return false;
+    }
+    if (comparesf2.__unordsf2(vector.a, vector.b) != vector.unReference) {
+        return false;
+    }
+    return true;
+}
+
+const arguments = []f32{
+    std.math.nan(f32),
+    -std.math.inf(f32),
+    -0x1.fffffep127,
+    -0x1.000002p0 - 0x1.000000p0,
+    -0x1.fffffep-1,
+    -0x1.000000p-126,
+    -0x0.fffffep-126,
+    -0x0.000002p-126,
+    -0.0,
+    0.0,
+    0x0.000002p-126,
+    0x0.fffffep-126,
+    0x1.000000p-126,
+    0x1.fffffep-1,
+    0x1.000000p0,
+    0x1.000002p0,
+    0x1.fffffep127,
+    std.math.inf(f32),
+};
+
+fn generateVector(comptime a: f32, comptime b: f32) TestVector {
```

Comment:
I had to make these function parameters comptime in order to use the function in the compile time initialization of the test_vectors array at line 91. The error said that the parameters needed to be comptime known, which they were because I was passing them in at compile time so I'm confused. I tried adding inline in front of the for loops and that didn't make a difference. Is this a bug?

In the documentation a function without comptime parameters is used at compile time but it is not contained within a block, which I suspect is the difference.

Reply:
For functions that return generics, the return type should be `type`.

Reply:
I don't think this function is supposed to return a generic? It's supposed to return a struct with the values filled out, not instantiate a struct type.

Am i understanding what you mean?

Reply:
Here is the relevant documentation: 

https://ziglang.org/documentation/master/#Compile-Time-Expressions

See the section:

```zig
const first_25_primes = firstNPrimes(25);
const sum_of_first_25_primes = sum(first_25_primes);
```

Reply:
Thanks for the fast replies -- misread `generateVector` as `genericVector`!

Reply:
>  Is this a bug?

That does sound like a bug. Before I merge this let me play around with it a bit and see if it's something I can fix along with the merge.